src/providers/http_provider.py

```diff
@@ -110,8 +111,9 @@ def _get_without_fallbacks(
         host: str,
         endpoint: str,
         path_params: Optional[Sequence[str | int]] = None,
-        query_params: Optional[dict] = None
-    ) -> Tuple[dict | list, dict]:
+        query_params: Optional[dict] = None,
+        stream: bool = False
+    ) -> Tuple[dict | list, dict] | Response:
```

Comment:
this return value like from different abstraction level
I'd prefer to process streaming here as it is with usual json response

Reply:
you also don't need cast everywhere in that case

Reply:
I would prefer to give the ability of handling the stream to the child implementations